Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.targets

```diff
@@ -17,68 +17,44 @@
   </PropertyGroup>
 
   <PropertyGroup>
     <CleanDependsOn>
       CleanFeatureFilesInProject;
+      CleanObsoleteCodeBehindFilesInProject;
       $(CleanDependsOn)
     </CleanDependsOn>
-    <RebuildDependsOn>
-      SwitchToForceGenerate;
-      $(RebuildDependsOn)
-    </RebuildDependsOn>
   </PropertyGroup>
 
   <!--
-    net.sdk support: update default compile items to show generated files as nested items
+    Show code-behind files as nested items under feature files in Visual Studio Solution Explorer
   -->
-  <ItemGroup Condition="'$(_ReqnrollEnableDefaultCompileItems)' == 'true' and '$(EnableDefaultItems)' == 'true' ">
-    <Compile Update="@(ReqnrollFeatureFiles->'%(CodeBehindFile)')"
-             DependentUpon="%(Filename)"
-             AutoGen="true"
-             DesignTime="true"
-             Visible="true"
-             Condition="'$(EnableDefaultCompileItems)' == 'true'" />
-
+  <ItemGroup>
+    <Compile Update="@(ReqnrollFeatureFiles->'%(CodeBehindFile)')" DependentUpon="%(Filename)" />
   </ItemGroup>
 
-
-  <Target Name="WarnForFeatureCodeBehindFilesWithoutCorrespondingFeatureFile" AfterTargets="CoreCompile"
-          Condition="'$(ReqnrollEnableWarnForFeatureCodeBehindFilesWithoutCorrespondingFeatureFile)' == 'true'">
-    <Warning Text="For codebehind file '@(ReqnrollObsoleteCodeBehindFiles)', no feature file was found." File="@(ReqnrollObsoleteCodeBehindFiles)" Condition="'@(ReqnrollObsoleteCodeBehindFiles)' != ''" />
+  <Target Name="WarnForObsoleteReqnrollCodeBehindFiles" AfterTargets="CoreCompile"
+          Condition="'$(ReqnrollWarnForObsoleteCodeBehindFiles)' == 'true' and '@(ReqnrollObsoleteCodeBehindFiles)' != ''">
+    <Warning Text="For code-behind file '@(ReqnrollObsoleteCodeBehindFiles)', no feature file was found. Set project property 'ReqnrollDeleteObsoleteCodeBehindFilesOnClean' to delete these files automatically on 'Clean' or 'Rebuild'." File="@(ReqnrollObsoleteCodeBehindFiles)" Condition="'$(ReqnrollDeleteObsoleteCodeBehindFilesOnClean)' != 'true'" />
+    <Warning Text="For code-behind file '@(ReqnrollObsoleteCodeBehindFiles)', no feature file was found. Perform a 'Clean' or 'Rebuild' operation to get these obsolete files deleted." File="@(ReqnrollObsoleteCodeBehindFiles)" Condition="'$(ReqnrollDeleteObsoleteCodeBehindFilesOnClean)' == 'true'" />
   </Target>
 
+  <Target Name="ProcessFeatureFilesInProject"
+          BeforeTargets="BeforeCompile;CoreCompile;AssignTargetPaths"
+          DependsOnTargets="PrepareForBuild;CoreProcessFeatureFilesInProject" />
 
-  <Target Name="SwitchToForceGenerate">
-    <PropertyGroup>
-      <ForceGeneration>true</ForceGeneration>
-    </PropertyGroup>
-  </Target>
-
-  <Target Name="UpdateFeatureFilesInProject"
-          DependsOnTargets="BeforeUpdateFeatureFilesInProject">
+  <Target Name="CoreProcessFeatureFilesInProject"
+          Inputs="@(ReqnrollFeatureFiles);$(MSBuildProjectFullPath);$(ReqnrollConfigFile)"
```

Comment:
What's the intention of including the project file in the inputs here?

Reply:
I think the project namespace is used too? Ultimately the file name just needs to vary when the inputs vary, so if you want to use the referenced Reqnroll package file, or the resolved assembly reference as an input, that's entirely valid.